### DISCUSSION TITLE
TimeSeriesDataFrame.convert_frequency SLOW !

### DISCUSSION THREAD

**Discussion by raedbsili1991** [autogluon/autogluon#4058] [Q&A]
Hello, 

I am working with irregular data with that I want to convert to monthly data set using the convert_frequency("M", agg_numeric = "sum"). 
Here is an exemple of what look like my TimeSeriesDataFrame 

![image](https://github.com/autogluon/autogluon/assets/31764213/9be49b21-9f0e-408a-9be1-493f1c674c14)

I wanted to infer the "Monthly" frequency before starting forecasting. 
However, the execution is "infintesily" slow! it never ended. 

I also tried to replicate the https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html but it the convert_frequency() is also so slow! 

[requirements.txt](https://github.com/autogluon/autogluon/files/14899071/requirements.txt) 

Why is this happening ?

**Comment by shchur** [comment]
Hi @raedbsili1991, thanks for reporting the problem. I have converted it into an issue (#4060) to better keep track of the progress.
